src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/UploadEnvelopeDocumentsTaskTest.java

```diff
@@ -75,8 +84,27 @@ void should_call_service_twice_when_2_different_containers_are_present_in_the_li
         verify(uploadService, times(2)).processByContainer(anyString(), anyList());
     }
 
-    private Envelope getEnvelope(String containerName) {
-        return new Envelope(
+    @Test
+    void should_not_send_envelopes_that_failed_too_many_times() {
+        // given
+        List<Envelope> envelopes = asList(
+            getEnvelope(CONTAINER_1, maxRetries - 1),
+            getEnvelope(CONTAINER_1, maxRetries),
+            getEnvelope(CONTAINER_1, maxRetries + 1)
+        );
+        given(envelopeRepository.findByStatusIn(asList(CREATED, UPLOAD_FAILURE)))
+            .willReturn(envelopes);
+
+        // when
+        task.run();
+
+        // then
+        verify(uploadService, times(1)).processByContainer(anyString(), envelopesCaptor.capture());
+        assertThat(envelopesCaptor.getValue()).hasSize(1);
```

Comment:
Might be useful to verify that it was the right envelope, given we already have a captor.